File: backend/app/schemas/matchmaking.py
from __future__ import annotations

import uuid
from datetime import datetime
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, ConfigDict, Field
from app.models.matchmaking import IndustryRelationshipType, MatchSuggestionStatus


class MatchingProfileBase(BaseModel):
    business_description: Optional[str] = None
    services_offered: List[str] = Field(default_factory=list)
    looking_for: List[str] = Field(default_factory=list)
    target_sectors: List[str] = Field(default_factory=list)
    matching_enabled: bool = True
    matching_preferences: Dict[str, Any] = Field(default_factory=dict)


class MatchingProfileCreate(MatchingProfileBase):
    pass


class MatchingProfileUpdate(MatchingProfileBase):
    pass


class MatchingProfile(MatchingProfileBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: uuid.UUID
    user_id: uuid.UUID
    created_at: datetime
    updated_at: datetime


class MatchSuggestionBase(BaseModel):
    user_id: uuid.UUID
    matched_user_id: uuid.UUID
    score: float
    score_breakdown: Dict[str, float] = Field(default_factory=dict)
    explanation: Optional[str] = None
    partnership_strategy: Optional[str] = None
    status: MatchSuggestionStatus = MatchSuggestionStatus.PENDING
    context: str = "general"


class MatchSuggestion(MatchSuggestionBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    expires_at: Optional[datetime] = None
    
    # Optionally include matched user summary
    matched_user_name: Optional[str] = None
    matched_user_photo: Optional[str] = None
    matched_user_industry: Optional[str] = None


class IndustryRelationshipBase(BaseModel):
    industry_a_id: uuid.UUID
    industry_b_id: uuid.UUID
    relationship_type: IndustryRelationshipType = IndustryRelationshipType.COMPLEMENTARY
    strength: float = 1.0


class IndustryRelationship(IndustryRelationshipBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: uuid.UUID

File: mobile/lib/models/matchmaking.dart
import 'package:json_annotation/json_annotation.dart';

part 'matchmaking.g.dart';

@JsonSerializable()
class MatchingProfile {
  final String id;
  @JsonKey(name: 'user_id')
  final String userId;
  @JsonKey(name: 'business_description')
  final String? businessDescription;
  @JsonKey(name: 'services_offered')
  final List<String> servicesOffered;
  @JsonKey(name: 'looking_for')
  final List<String> lookingFor;
  @JsonKey(name: 'target_sectors')
  final List<String> targetSectors;
  @JsonKey(name: 'matching_enabled')
  final bool matchingEnabled;

  MatchingProfile({
    required this.id,
    required this.userId,
    this.businessDescription,
    required this.servicesOffered,
    required this.lookingFor,
    required this.targetSectors,
    required this.matchingEnabled,
  });

  factory MatchingProfile.fromJson(Map<String, dynamic> json) => _$MatchingProfileFromJson(json);
  Map<String, dynamic> toJson() => _$MatchingProfileToJson(this);
}

@JsonSerializable()
class MatchSuggestion {
  final String id;
  @JsonKey(name: 'matched_user_id')
  final String matchedUserId;
  final double score;
  @JsonKey(name: 'score_breakdown')
  final Map<String, double> scoreBreakdown;
  final String? explanation;
  @JsonKey(name: 'partnership_strategy')
  final String? partnershipStrategy;
  final String status;
  
  @JsonKey(name: 'matched_user_name')
  final String? matchedUserName;
  @JsonKey(name: 'matched_user_photo')
  final String? matchedUserPhoto;
  @JsonKey(name: 'matched_user_industry')
  final String? matchedUserIndustry;

  MatchSuggestion({
    required this.id,
    required this.matchedUserId,
    required this.score,
    required this.scoreBreakdown,
    this.explanation,
    this.partnershipStrategy,
    required this.status,
    this.matchedUserName,
    this.matchedUserPhoto,
    this.matchedUserIndustry,
  });

  factory MatchSuggestion.fromJson(Map<String, dynamic> json) => _$MatchSuggestionFromJson(json);
  Map<String, dynamic> toJson() => _$MatchSuggestionToJson(this);
}
